### DISCUSSION TITLE
Line width

### DISCUSSION THREAD

**Discussion by raul-parada** [shapely/shapely#1572] [Q&A] [answered]
I would like to adapt the LineString width according to a given real world width. Which is the default width in real world of the LineString? I'm tring to adapt it doing line.crs='EPSG:3857'.
However, I get this warning.

ShapelyDeprecationWarning: Setting custom attributes on geometry objects is deprecated, and will raise an AttributeError in Shapely 2.0
  line.crs='EPSG:3857'.

For instance, I would like to generate a line with real world width of 2.5 meters and another with 1.5 meters. How can I do it? 
Thanks.

**Comment by jorisvandenbossche** [comment, marked as answer]
@raul-parada a LineString itself doesn't have a "width" (if you want an object with an actual area, you would need to use a Polygon). It's only when visualizing a LineString that you typically use a certain width to represent the linestring on the screen. 

What kind of visualization are you thinking about in this case? 
If on the other hand you want an actual object with a width, you can use `buffer()` to convert a LineString to a Polygon with a certain width.

**Comment by raul-parada** [comment]
Well, it is not visualization, it is about if two lines intercept or not. I'm using this code:

```
from shapely.geometry import LineString

line = LineString([(val[0][0], val[1][0]), (val[0][len(val)-1], val[1][len(val)-1])])
other = LineString([(val1[0][0], val1[1][0]), (val1[0][len(val1)-1], val1[1][len(val1)-1])])
print(line.intersects(other))
```

These lines represent the trajectories of two vehicles, with different width. Then, the closer the width with reality, more realistic will be the use case to detect that intersection. How can I use the `buffer()` function?

**Reply by jorisvandenbossche** [reply]
In that case, converting to Polygons seems the correct thing to do.

> How can I use the buffer() function?

See the manual: https://shapely.readthedocs.io/en/stable/manual.html#object.buffer

**Comment by raul-parada** [comment]
Which is the default "width" of the line? Is .buffer(0.5) a kind of factor to multiply the width of the line? 
For instance, in my case, `line.buffer(0.5)`
Right?

**Reply by jorisvandenbossche** [reply]
As I mentioned before, a line does not have any "width". A LineString is a collection of connected points, it does not have an area. 

So the distance you pass to `buffer` determines the size of the resulting polygon. If you pass 0.5 (which is the radius), you will get a polygon of width 1 (around the straight parts of the line), and this 1 is in the units of your coordinates.
